src/zkevm_specs/evm/instruction.py

```diff
@@ -74,32 +74,8 @@ def constrain_equal(self, lhs: int, rhs: int):
     def constrain_bool(self, value: int):
         assert value in [0, 1]
 
-    def constrain_transfer(
-        self,
-        sender_address: int,
-        receiver_address: int,
-        value: int,
-        gas_fee: int = 0,
-        is_persistent: bool = True,
-        rw_counter_end_of_reversion: int = 0,
-    ):
-        sender_balance, sender_balance_prev = self.account_write_with_reversion(
-            sender_address, AccountFieldTag.Balance, is_persistent, rw_counter_end_of_reversion
-        )
-        receiver_balance, receiver_balance_prev = self.account_write_with_reversion(
-            receiver_address, AccountFieldTag.Balance, is_persistent, rw_counter_end_of_reversion
-        )
-
-        value_with_gas_fee, overflow = self.add_word(value, gas_fee)
-        self.constrain_zero(overflow)
-
-        result, carry = self.add_word(value_with_gas_fee, sender_balance)
-        self.constrain_equal(sender_balance_prev, result)
-        self.constrain_zero(carry)
-
-        result, carry = self.add_word(value, receiver_balance_prev)
-        self.constrain_equal(receiver_balance, result)
-        self.constrain_zero(carry)
+    def constrain_sufficient_gas_left(self, gas_left: int):
+        self.int_to_bytes(gas_left, N_BYTES_GAS)
```

Comment:
Is the intention of this function an underflow check?

Reply:
Yes

Reply:
What do you think we rename `constrain_sufficient_gas_left ` to something like `constrain_not_underflow` to reflect the purpose?

Reply:
Sounds good to me. Renamed in [`6b37c0b`](https://github.com/appliedzkp/zkevm-specs/pull/78/commits/6b37c0b7f5e755c13f44642b8f04f27c13483ee9)